pkg/querier/queryrange/split_by_interval.go

```diff
@@ -276,25 +276,35 @@ func splitByTime(req queryrangebase.Request, interval time.Duration) ([]queryran
 	return reqs, nil
 }
 
-// maxRangeVectorDuration returns the maximum range vector duration within a LogQL query.
-func maxRangeVectorDuration(q string) (time.Duration, error) {
-	expr, err := syntax.ParseSampleExpr(q)
+// maxRangeVectorAndOffsetDuration returns the maximum range vector and offset duration within a LogQL query.
+func maxRangeVectorAndOffsetDuration(q string) (time.Duration, time.Duration, error) {
+	expr, err := syntax.ParseExpr(q)
 	if err != nil {
-		return 0, err
+		return 0, 0, err
+	}
+
+	if _, ok := expr.(syntax.SampleExpr); !ok {
+		return 0, 0, nil
 	}
-	var max time.Duration
+	
+	var maxRVDuration, maxOffset time.Duration
 	expr.Walk(func(e interface{}) {
-		if r, ok := e.(*syntax.LogRange); ok && r.Interval > max {
-			max = r.Interval
+		if r, ok := e.(*syntax.LogRange); ok {
+			if r.Interval > maxRVDuration {
```

Comment:
Q: whats the point of comparing it with `maxRvDuration` and `maxOffset`. They have only zero values right? (from the above declarations)

why not just `maxRVDuration, maxOffset = r.Internval, r.Offset`? Given, there can be only one `LogRange` expression in a given LogQL.? (can LogQL contains multiple Range expression? I doubt it though)

Reply:
Gotcha! :+1: